src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,365 @@
+
+For example:
+
+```cpp
+// Define a function in namespace scope
+bool is_even(int x) {
+	return x % 2 == 0;
+}
+
+int main() {
+	std::vector<int> data = { 1, 3, 5, 10, 73 };
+
+	// Actually use the function here
+	bool has_even = std::ranges::any_of(data, is_even);
+}
+```
+
+can instead be rewritten as:
+
+```cpp
+int main() {
+	std::vector<int> data = { 1, 3, 5, 10, 73 };
+
+	// Define a lambda in-line
+	bool has_even = std::ranges::any_of(data, [](int x) { return x % 2 == 0; });
+}
+```
+
+&nbsp;
+
+&nbsp;
+
+## Syntax
+
+The basic syntax of a lambda expression looks like this:
+
+```
+[captures](params) -> ReturnType {
+	statements;
+}
+```
+
+A more detailed explanation of the syntax can be found on
+[cppreference](https://en.cppreference.com/w/cpp/language/lambda#Syntax).
+
+&nbsp;
+
+Each lambda expression has its own unique, unnameable type:
+
+```cpp
+auto add = [](int x, int y) { return x + y; };
+```
+
+is similar to writing:
+
+```cpp
+struct {
+	auto operator()(int x, int y) const {
+		return x + y;
+	}
+} add;
+```
+
+This also means that even identical lambda expressions always have completely different types, and are not
+interconvertible:
+
+```cpp
+auto foo = [](int x) { return x + 1; };
+auto bar = [](int x) { return x + 1; };
+
+static_assert(not std::same_as<decltype(foo), decltype(bar)>);
+```
+
+&nbsp;
+
+### Capture list
+
+All lambda expressions begin with a capture list. The capture list specifies which variables to capture from the
+surrounding scope:
+
+```cpp
+int x = 10;
+
+auto lambda = [x](int y) { return x + y; };
+
+lambda(5) // 15
+```
+
+- `[]` - Empty capture list.
+- `[=]` - Automatically capture variables that are used in the lambda body by value.
+  - Mutually exclusive with `[&]`.
+  - Does not implicitly capture `this` if used in a class.
+- `[&]` - Automatically capture variables that are used in the lambda body by reference.
+  - Mutually exclusive with `[=]`.
+- `[x]` - Capture only `x` by value.
+- `[&x]` - Capture only `x` by reference.
+- `[x...]` - Capture a pack `x` by value.
+- `[&x...]` - Capture a pack `x` by reference.
+- `[this]` - Capture `*this` by reference.
+- `[x = y]` - Define a local variable `x` and initialize it to `y`.
+
+Different captures can be mixed together:
+
+```cpp
+int a = 1;
+int b = 2;
+int c = 3;
+
+auto foo = [=, &b, d = c]() {
+	// `a` is implicitly captured by value,
+	// `b` is explicitly captured by reference, and
+	// `d` is initialized to `c`.
+	return a + b + d;
+};
+```
+
+However, `[=]` may only be followed by captures by reference and `[&]` may only be followed by captures by value.
+
+Implicit captures can be nested multiple times:
+
+```cpp
+int x = 0;
+
+[&]() {
+	[&]() {
+		x = 5;
+	}();
+}();
+```
+
+&nbsp;
+
+### Parameters
+
+Lambda parameters work the same way as normal function parameters, and `auto` parameters make a lambda's `operator()`
+implicitly templated.
+
+If a lambda takes no parameters, the parameter list may be omitted entirely:
```

Comment:
I'd probably move this to the end of the section, just so things flow more naturally (discussion about parameters followed by an example of parameters instead of a discussion of parameters followed by an example of no parameters)